scalarmult_test.go

```diff
@@ -93,6 +93,28 @@ func TestScalarMulDistributesOverAdd(t *testing.T) {
 	}
 }
 
+func TestScalarMulNonIdentityPoint(t *testing.T) {
+	// Check whether p.ScalarMult and q.ScalaBaseMult give the same,
+	// when p and q are originally set to the base point.
+
+	scalarMulNonIdentityPoint := func(x Scalar) bool {
+		var p, q Point
```

Comment:
Even better, you can randomize the receivers, by taking them as arguments to the quickcheck function.

I guess at that point the test name would be `TestScalarMultLikeScalarBaseMult`, and it would happen to also test non-identity receivers.